<?xml version="1.0" encoding="UTF-8"?>
<launch>
  <arg name="filename" default="default"/>
  <arg name="path" default="/home/andrew/tests/data/subway"/>
  <arg name="model_name" default="X4"/>

  <!-- Launch bag listener  -->
  <group ns="$(arg model_name)">
    <node pkg="test_sim" type="save_test_data" name="save_test_data" output="screen" required="true">
        <param name="rate" value="2.0" />
        <param name="filename_map" value="$(arg path)/$(arg filename)_map.csv" />
        <param name="filename_poses" value="$(arg path)/$(arg filename)_poses.csv" />
        <param name="filter_by_bbox" value="true"/>
        <param name="bbox_min_z" value = "0.0"/>
        <param name="bbox_max_z" value = "3.0"/>
        <param name="bbox_min_y" value = "-40.0"/>
        <param name="bbox_max_y" value = "40.0"/>
        <param name="bbox_min_x" value = "-14.0"/>
        <param name="bbox_max_x" value = "14.0"/>
        <remap from="odometry" to="odometry_throttle"/>
        <remap from="octomap_binary" to="octomap_binary_throttle"/>
    </node>
  </group>

  <!-- Launch bag node -->
  <node pkg="rosbag" type="play" name="rosbag_play_test" args="$(arg path)/$(arg filename).bag -r 6"/>
</launch>